setup.py

```diff
@@ -77,11 +77,16 @@ def build_extension(self, ext: CMakeExtension) -> None:
         # Pass version to C++
         cmake_args += [f"-DMLX_VERSION={self.distribution.get_version()}"]  # type: ignore[attr-defined]
 
-        if sys.platform.startswith("darwin"):
+        if platform.system() == "Darwin":
```

Comment:
Looks good overall.. but I'm just curious what if anything motivated the switch to `platform.system` from `sys.platform`?

Reply:
MLX's codebase is using both `platform.system` and `sys.platform` but more code is using the former, so I chose the former for the new code and made this file use the same style.